test/Objects/Schema/Generation/SchemaGeneratorTest.php

```diff
@@ -45,6 +45,7 @@ public function it_should_generate_an_empty_record()
             ->namespace('org.acme');
 
         $this->assertEquals($expected, $schema);
+        $schema->parse();
```

Comment:
Why do we need to parse the schema after the expectation? Is there maybe another test assumption/condition missing?

Reply:
I was not so sure also if it was the best place to do this here. My intention is just to make sure the generated schema object is parseable. There are some checks and validations that will be applied only after we actual do the parsing.

So if anything is wrong, an exception would be raised at this point.

Another idea I had was to create a data provider that would provide the class with the annotations and the expected schema. Then we would have two tests: One to actually see if the generated schema matches the expected one and the other to make sure the generated schema parses without errors.

What do you think?

Reply:
I believe that testing for the failure conditions should be a separate test. That test can use a data provider to provide invalid edge cases and make sure that an exception is thrown. Right now I believe the tests are mixing up different responsibilities - testing that no exception is thrown and the valid parsing of annotations.

Reply:
Idd. Also in this case testing for failures and expect exceptions wouldn't make much sense. Because the parsing and validation of schema is responsibility of the avro-php package. I'm going to remove these "parse" calls